phoenix-core/src/main/java/org/apache/hadoop/hbase/ipc/controller/ServerSideRPCControllerFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.ipc.controller;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.ipc.RpcControllerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link RpcControllerFactory} that should only be used when
+ * making server-server remote RPCs to the region servers hosting Phoenix SYSTEM tables.
```

Comment:
@jpisaac - if this is only for server-to-server RPCs to system tables, how is this different from the existing metadata pool? I thought that's what that was already for?

Reply:
@gjacoby126 The MetadataRpcController implements the org.apache.hbase.thirdparty.com.google.protobuf.RpcController which is not compatible with ServerRpcController which implements com.google.protobuf.RpcController. Hence had to use a separate executor for server-server RPC to Metadata tables.

Reply:
OK, so let me make sure I understnad properly. 

Direct operations on a system table (UPSERTs or queries) would use the MetadataRpcController and coproc calls (such as CREATE and ALTER operations where MetadataClient calls MetadataEndpointImpl operations) would use ServerSideRPCController? And there's no way to consolidate the two in either direction because of the mismatch between shaded and unshaded protobufs disallows casting?